packages/example/tests/playwright/specs/cookies.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { LoginForm } from "../models/login-form";
+import { test } from "../test";
+
+// NOTE: tests are intentionally sequential, do not use `test.describe.parallel` here
+test.describe.serial("cookies", () => {
+  test("should not have a session cookie if the user has not logged in yet", async ({
+    page,
+  }) => {
+    await page.goto("/login");
+
+    const loginForm = new LoginForm(page);
+    await loginForm.assertSessionStatus(401);
+  });
+
+  test("should have a valid session once the user logs in", async ({
+    page,
+  }) => {
+    await page.goto("/login");
+
+    const loginForm = new LoginForm(page);
+    await loginForm.loginWithValidCredentials();
+    await loginForm.assertSessionStatus(200);
+  });
+
+  test("should still have a valid session if the user refreshes the page after logging in", async ({
+    page,
+  }) => {
+    await page.goto("/login");
+
+    const loginForm = new LoginForm(page);
+    await loginForm.loginWithValidCredentials();
+
+    // Reload the page
+    await page.reload();
+
+    await loginForm.assertSessionStatus(200);
+  });
+
+  test("should not have a session from the previous test run", async ({
```

Comment:
I think it makes sense to move it to a place after the first test with successful login. Otherwise, in case it doesn't clear cookies, this test won't be run because "should still have a valid session if the user refreshes the page after logging in" will fail instead.

Another option is to wrap two tests in a group (in that case we can make other tests parallel):

```ts
test.describe.serial("when running sequentially", () => {
  test("should login", () => { ... });
  
  test("should not have a session from the previous test run", () => { ... });
});
```

Reply:
I really like both options. I'm going to go with the second one since it has a performance benefit as well 😎